ui/stage/stage-test.tsx

```diff
@@ -42,7 +33,9 @@ function useTestData(device: HIDDevice | undefined) {
     const d = device;
     async function update() {
       const data = await requestTestData(d);
-      setTestData(data);
+      if (data) {
+        setTestData(data);
+      }
```

Comment:
This seems to work for the most part, but I still get some console errors when I interact with the UI buttons for sending test data and I'm not too sure why.

Reply:
I think with the way things are currently setup, we're always going to get some weird in-fighting between the test command buttons and the automatic reads happening here. It won't be a problem in the long term, so this should be fine to just ignore.